### PR TITLE
fix(Video): improve perfomance and stability

### PR DESCRIPTION
**Pull request by marcoskolodny**
Stability and performance fixes in Video component.

This PR fixes the following issues (tested in web, Android and iOS):
- https://jira.tid.es/browse/WEB-1439
- https://jira.tid.es/browse/TGT-2149
- https://jira.tid.es/browse/TGT-2277
- https://jira.tid.es/browse/NOVUMCC-8351

**Note: after merging this PR and releasing a new Mistica version, we should make a small change in webapp to adapt the code to these changes.**

### CODE REVIEW

File: src/video.tsx [hunk 1]
```diff
@@ -238,8 +248,47 @@ const Video = React.forwardRef<HTMLVideoElement, VideoProps>(
                 width={props.width}
                 height={props.height}
             >
+                <div
+                    style={{
+                        position: 'absolute',
+                        width: '100%',
+                        height: '100%',
+                    }}
+                    ref={(element) => {
```
Comment (atabel) [inline src/video.tsx:257]: shouldn't this ref be in <video> ?

File: src/video.tsx [hunk 2]
```diff
@@ -238,8 +248,47 @@ const Video = React.forwardRef<HTMLVideoElement, VideoProps>(
...
+                        const containerElement = element ? (element as VideoElement) : null;
+
+                        if (containerElement) {
+                            containerElement.play = () => videoRef.current?.play() || new Promise(() => {});
```
Comment (atabel) [inline src/video.tsx:261]: why do you return a never resolving promise here?

File: src/video.tsx (hunk 1 above)
Comment (atabel) [inline src/video.tsx:257]: I'd include a code comment explaining why this ref logic is needed

**Review by atabel — COMMENTED** [review]
We should verify that this fixes all these issues:
https://jira.tid.es/browse/WEB-1439
https://jira.tid.es/browse/TGT-2149
https://jira.tid.es/browse/TGT-2277
https://jira.tid.es/browse/NOVUMCC-8351

Also, can we cover some of these cases with acceptance tests?

File: src/video.tsx (hunk 2 above)
Comment (marcoskolodny) [reply src/video.tsx]: fixed

File: src/video.tsx (hunk 1 above)
Comment (atabel) [reply src/video.tsx:262]: ok, it's now explained with the code comment

**Review by atabel — APPROVED** [review]

File: src/video.tsx [hunk 3]
```diff
@@ -266,8 +272,18 @@ const Video = React.forwardRef<VideoElement, VideoProps>(
...
+                                 * The issue happened when calling .load() form the component's exposed ref
+                                 */
+                                if (posterRef.current?.style) {
+                                    posterRef.current.style.width = '100%';
+                                    posterRef.current.style.height = '100%';
+                                }
+                                setTimeout(() => {
+                                    dispatch('reset');
+                                    videoRef.current?.load();
+                                }, 100);
```
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/video.tsx:294]: done

**Review by pladaria — APPROVED** [review]
nice work! thanks for the comments :)

**Review by atabel — APPROVED** [review]

**Pull request merged**